os/services/rpl-border-router/native/slip-config.c

```diff
@@ -197,14 +111,112 @@ slip_config_handle_arguments(int argc, char **argv)
-    break;
+    fprintf(stderr, "unknown baudrate %s", optarg);
+    return 1;
+  }
+  return 0;
+}
+CONTIKI_OPTION(BAUDRATE_PRIO, { "B", required_argument, NULL, 0 },
+               baudrate_callback,
+#ifdef linux
+               "baudrate (9600,19200,38400,57600,115200,921600)"
+#else
+               "baudrate (9600,19200,38400,57600,115200)"
+#endif
+               " (default 115200)\n");
+CONTIKI_OPTION(BAUDRATE_PRIO + 1,
+               { "H", no_argument, &slip_config_flowcontrol, 1 }, NULL,
+               "hardware CTS/RTS flow control (default disabled)\n");
+CONTIKI_OPTION(BAUDRATE_PRIO + 2,
+               { "L", no_argument, &slip_config_timestamp, 1 }, NULL,
+               "log output format (adds time stamps)\n");
+static int
+device_callback(const char *optarg)
+{
+  slip_config_siodev = optarg + (strncmp("/dev/", optarg, 5) == 0 ? 5 : 0);
+  return 0;
+}
+CONTIKI_OPTION(BAUDRATE_PRIO + 3, { "s", required_argument, NULL, 0 },
+               device_callback, "serial device (default /dev/ttyUSB0)\n");
+static int
+host_callback(const char *optarg)
+{
+  slip_config_host = optarg;
+  return 0;
+}
+CONTIKI_OPTION(BAUDRATE_PRIO + 4, { "a", required_argument, NULL, 0 },
+               host_callback, "connect via TCP to server at <value>\n");
+static int
+port_callback(const char *optarg)
+{
+  slip_config_port = optarg;
+  return 0;
+}
+CONTIKI_OPTION(BAUDRATE_PRIO + 5, { "p", required_argument, NULL, 0 },
+               port_callback, "connect via TCP to server on port <value>\n");
+static int
+dev_callback(const char *optarg)
+{
+  strncpy(slip_config_tundev,
+          optarg + (strncmp("/dev/", optarg, 5) == 0 ? 5 : 0),
+          sizeof(slip_config_tundev) - 1);
+  slip_config_tundev[sizeof(slip_config_tundev) - 1] = '\0';
+  return 0;
+}
+CONTIKI_OPTION(BAUDRATE_PRIO + 6, { "t", required_argument, NULL, 0 },
+               dev_callback, "name of interface (default tun0)\n");
+static int
+delay_callback(const char *optarg)
+{
+  slip_config_basedelay = optarg ? atoi(optarg) : 10;
+  if(slip_config_basedelay < 0 ||
+     (slip_config_basedelay == 0 && optarg && optarg[0] != '0')) {
+    fprintf(stderr, "Delay '%s' could not be parsed as a number\n", optarg);
+    return 1;
   }
 
-  if(*slip_config_tundev == '\0') {
-    /* Use default. */
-    strcpy(slip_config_tundev, "tun0");
+  return 0;
+}
+CONTIKI_OPTION(BAUDRATE_PRIO + 7, { "d", optional_argument, NULL, 0 },
+               delay_callback,
+               "minimum delay between outgoing SLIP packets (default 10)\n"
+               "\t\tActual delay is basedelay * (#6LowPAN fragments)"
+               " milliseconds.\n");
+/*---------------------------------------------------------------------------*/
+int
+slip_config_handle_arguments(int argc, char **argv)
+{
+  if(argc != 2 && argc != 3) {
+    err(1, "usage: %s [-B baudrate] [-H] [-L] [-s siodev] [-t tundev] [-T] [-v verbosity] [-d delay] [-a serveraddress] [-p serverport] ipaddress", argv[0]);
```

Comment:
This prints the previous argument as application name:

`./build/native/border-router.native -s /dev/ttyACM0`
...
`border-router.native: usage: /dev/ttyACM0 [-B baudrate] [-H] [-L] [-s siodev] [-t tundev] [-T] [-v verbosity] [-d delay] [-a serveraddress] [-p serverport] ipaddress: No such file or directory`

Perhaps simply remove the print of `argv[0]` ?

Reply:
You're right. Fixed.